### PR TITLE
CME-23 upgrade to Spring Boot 3

### PR DESCRIPTION
**Pull request by rapolaskaseliscgi**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/CME-23

### Change description ###
Update to Spring Boot 3

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/bulkscanning/controller/PaymentControllerTest.java [hunk 1]
```diff
@@ -145,7 +145,7 @@ public void testCreatePaymentForBulkScan() throws Exception{
        when(paymentService.saveInitialMetadataFromBs(any(BulkScanPaymentRequest.class)))
            .thenReturn(Arrays.asList(dcn));
 
-       ResultActions resultActions = mockMvc.perform(post("/bulk-scan-payments/")
+       ResultActions resultActions = mockMvc.perform(post("/bulk-scan-payments")
```
Comment (rapolaskaseliscgi) [inline src/test/java/uk/gov/hmcts/reform/bulkscanning/controller/PaymentControllerTest.java:148]: Pay attention to this potentially breaking change. In V2 those two urls were treated the same, in V3, they are different and therefore we need to be explicit if we want to handle urls ending with `/` the same as without slash.

Comment (kiran-yenigala-hmcts) [reply src/test/java/uk/gov/hmcts/reform/bulkscanning/controller/PaymentControllerTest.java:148]: yes, this is correct implementation, I have seen this behaviour on Ref Data

**Review by kiran-yenigala-hmcts — APPROVED** [review]

File: src/test/java/uk/gov/hmcts/reform/bulkscanning/controller/PaymentControllerTest.java (hunk 1 above)
Comment (rapolaskaseliscgi) [reply src/test/java/uk/gov/hmcts/reform/bulkscanning/controller/PaymentControllerTest.java:148]: What I mean breaking is that if clients submitting with `/` at the end of url, with spring boot 2, it simply worked, with the upgrade, it is going to break for them.

Comment (davejones74) [reply src/test/java/uk/gov/hmcts/reform/bulkscanning/controller/PaymentControllerTest.java:148]: Thanks @rapolaskaseliscgi POST /bulk-scan-payments is the correct format.

File: src/test/java/uk/gov/hmcts/reform/bulkscanning/functionaltest/PaymentControllerFnTest.java [hunk 2]
```diff
@@ -262,9 +263,9 @@ public void testNonMatchingPaymentsFromExelaThenBulkScan() throws Exception {
 
     @Test
     @WithMockUser(authorities = "payments")
-    public void testMatchingBulkScanFirstThenExela() throws Exception {
+    public void testMatchingBulkScanFirstThenExela() {
         //Request from Bulk Scan with one DCN
-        String dcn[] = {"111122223333888811111", "111122223333999911111"};
+        String[] dcn = {"111122223333888811111", "111122223333999911111"};
```
Comment (davejones74) [inline src/test/java/uk/gov/hmcts/reform/bulkscanning/functionaltest/PaymentControllerFnTest.java:268]: This is definately more readable, thank you. Syntatically it doesn't really matter though, but a good style change.

File: src/test/java/uk/gov/hmcts/reform/bulkscanning/model/request/BulkScanPaymentRequestTest.java [hunk 3]
```diff
@@ -57,7 +57,7 @@ public void testInvalidCcdCaseNumber(){
             fail("should have thrown an Error Message on ccd_case_number");
         }else {
             violations.stream().forEach(v -> {
-                if (v.getMessage().equals("ccd_case_number should be numeric")) {
+                if ("ccd_case_number should be numeric".equals(v.getMessage())) {
```
Comment (davejones74) [inline src/test/java/uk/gov/hmcts/reform/bulkscanning/model/request/BulkScanPaymentRequestTest.java:60]: Null pointer safe, nice one.

**Pull request merged**